src/element/mutateElement.ts

```diff
@@ -12,29 +12,28 @@ type ElementUpdate<TElement extends ExcalidrawElement> = Omit<
 // The version is used to compare updates when more than one user is working in
 // the same drawing. Note: this will trigger the component to update. Make sure you
 // are calling it either from a React event handler or within unstable_batchedUpdates().
-export function mutateElement<TElement extends ExcalidrawElement>(
+export function mutateElement<TElement extends Mutable<ExcalidrawElement>>(
   element: TElement,
   updates: ElementUpdate<TElement>,
 ) {
-  const mutableElement = element as any;
-
   for (const key in updates) {
     const value = (updates as any)[key];
     if (typeof value !== "undefined") {
-      mutableElement[key] = value;
+      // @ts-ignore
+      element[key] = value;
     }
   }
 
   if (
     typeof updates.height !== "undefined" ||
     typeof updates.width !== "undefined" ||
-    typeof updates.points !== "undefined"
+    typeof (updates as any).points !== "undefined"
```

Comment:
Not sure what hack to use here. In this case, doing `"points" in updates && updates.points !== "undefined"` makes `updates` into `never` type in the right-hand expression, so can't do that. See https://github.com/microsoft/TypeScript/issues/21732

Reply:
Does `isLinearElement(update) && typeof updates.points !== "undefined"` work?

Reply:
No, because `update` is a partial that doesn't contain `type` (union determinant).

Reply:
does `isLinearElement(element) && typeof updates.points !== "undefined"` work?

if not, i think this hack is fine.